File: lib/java/openlinktoken/src/main/java/org/openlinktoken/tokens/tokenizer/SHA256TokenDigest.java
/* SPDX-License-Identifier: MIT */
package org.openlinktoken.tokens.tokenizer;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

import org.openlinktoken.crypto.CryptoSuite;

/**
 * Calculates SHA-256 token digests.
 */
public final class SHA256TokenDigest implements TokenDigest {
    private static final long serialVersionUID = 1L;

    /**
     * Calculates a SHA-256 digest.
     *
     * @param value the token-signature bytes
     * @return the SHA-256 digest
     * @throws NoSuchAlgorithmException if SHA-256 is unavailable
     */
    @Override
    public byte[] digest(byte[] value) throws NoSuchAlgorithmException {
        return MessageDigest.getInstance(CryptoSuite.TOKEN_DIGEST_SHA256).digest(value);
    }
}

File: lib/python/openlinktoken/src/main/openlinktoken/tokens/tokenizer/sha256_token_digest.py
# SPDX-License-Identifier: MIT
"""SHA-256 token digest implementation."""

import hashlib

from openlinktoken.tokens.tokenizer.token_digest import TokenDigest


class Sha256TokenDigest(TokenDigest):
    """Calculate SHA-256 digests for token-signature bytes.

    Args:
        None.

    Returns:
        A ``Sha256TokenDigest`` instance.
    """

    def digest(self, value: bytes) -> bytes:
        """Return the SHA-256 digest of supplied bytes.

        Args:
            value: Token-signature bytes to digest.

        Returns:
            The 32-byte SHA-256 digest.
        """
        return hashlib.sha256(value).digest()
